src/shared/istiopolicy/creator.go

```diff
@@ -0,0 +1,181 @@
+		if err != nil {
+			c.recorder.RecordWarningEventf(intents, ReasonCreatingIstioPolicyFailed, "could not create istio policies: %s", err.Error())
+			return err
+		}
+	}
+
+	if len(intents.GetCallsList()) > 0 {
+		c.recorder.RecordNormalEventf(intents, ReasonCreatedIstioPolicy, "istio policies reconcile complete, reconciled %d servers", len(intents.GetCallsList()))
+	}
+
+	return nil
+}
+
+func (c *Creator) namespaceNotAllowed(intent v1alpha2.Intent, requestNamespace string) bool {
+	targetNamespace := intent.GetServerNamespace(requestNamespace)
+	restrictedNamespacesExists := len(c.restrictToNamespaces) != 0
+	return restrictedNamespacesExists && !lo.Contains(c.restrictToNamespaces, targetNamespace)
+}
+
+func (c *Creator) updateOrCreatePolicy(
+	ctx context.Context,
+	intents *v1alpha2.ClientIntents,
+	intent v1alpha2.Intent,
+	objectNamespace string,
+	clientServiceAccountName string,
+) error {
+	clientName := intents.Spec.Service.Name
+	policyName := fmt.Sprintf(OtterizeIstioPolicyNameTemplate, intent.GetServerName(), clientName)
+	newPolicy := c.getAuthorizationPolicyForIntent(intent, objectNamespace, policyName, clientServiceAccountName)
+
+	existingPolicy := &v1beta1.AuthorizationPolicy{}
+	err := c.client.Get(ctx, types.NamespacedName{
+		Name:      policyName,
+		Namespace: intent.GetServerNamespace(objectNamespace)},
+		existingPolicy)
+	if err != nil && !errors.IsNotFound(err) {
+		c.recorder.RecordWarningEventf(existingPolicy, ReasonGettingIstioPolicyFailed, "failed to get istio policy: %s", err.Error())
+		return err
+	}
+
+	if errors.IsNotFound(err) {
+		err = c.client.Create(ctx, newPolicy)
+		if err != nil {
+			c.recorder.RecordWarningEventf(existingPolicy, ReasonCreatingIstioPolicyFailed, "failed to istio policy: %s", err.Error())
+			return err
+		}
+		return nil
+	}
+
+	logrus.Infof("Found existing istio policy %s", policyName)
+
+	if !c.isPolicyEqual(existingPolicy, newPolicy) {
+		logrus.Infof("Updating existing istio policy %s", policyName)
+		policyCopy := existingPolicy.DeepCopy()
+		policyCopy.Spec.Rules[0].From[0].Source.Principals[0] = newPolicy.Spec.Rules[0].From[0].Source.Principals[0]
+		policyCopy.Spec.Selector.MatchLabels[v1alpha2.OtterizeServerLabelKey] = newPolicy.Spec.Selector.MatchLabels[v1alpha2.OtterizeServerLabelKey]
+		err = c.client.Patch(ctx, policyCopy, client.MergeFrom(existingPolicy))
+		if err != nil {
+			c.recorder.RecordWarningEventf(existingPolicy, ReasonUpdatingIstioPolicyFailed, "failed to update istio policy: %s", err.Error())
+			return err
+		}
+	}
+
+	return nil
+}
+
+func (c *Creator) isPolicyEqual(existingPolicy *v1beta1.AuthorizationPolicy, newPolicy *v1beta1.AuthorizationPolicy) bool {
+	sameServer := existingPolicy.Spec.Selector.MatchLabels[v1alpha2.OtterizeServerLabelKey] == newPolicy.Spec.Selector.MatchLabels[v1alpha2.OtterizeServerLabelKey]
+	samePrincipal := existingPolicy.Spec.Rules[0].From[0].Source.Principals[0] == newPolicy.Spec.Rules[0].From[0].Source.Principals[0]
```

Comment:
Well, I did it because we are creating this object so I have to confidence to assume they exist. If someone will manually creates a file in the same name and won't create rules or from we will crush. So I'll check it.